### PR TITLE
Sentiment data

### PR DESCRIPTION
**Pull request by MynahMarie**
In this pull request I did:

* For each article in the list displayed, added a hidden field in the handlebars rendering template that contains the id of each article. 
* Each visualize button makes an api call to retrieve the full bodyText of that given id.
* The data for now is only returned in the console. It is ready to be processed with D3 on front end.
* Added a morgan logger so that it's easier to debug the code.
* Changed the npm scripts so that nodemon runs with `npm run dev` (because at some point we will want to push to Heroku and Heroku needs the start script to be with node).

# Relates to #12

### CODE REVIEW

File: public/main.js [hunk 1]
```diff
@@ -1,16 +1,33 @@
...
+  xhr.send(JSON.stringify(data));
 }
 
+var buttons = document.querySelectorAll('.article__btn');
 
+buttons.forEach(function(btn) {
+  btn.addEventListener('click', function(e) {
+    e.preventDefault();
+    var query = [btn.parentElement[0].value];
+    console.log(query);
```
Comment (claireinez) [inline public/main.js]: Get rid of the console.log

File: public/main.js [hunk 2]
```diff
@@ -1,16 +1,33 @@
...
+    } else if (xhr.readyState == 4 && xhr.status === 200) {
+      callback(null, xhr.responseText);
+    }
+  };
+  xhr.open('POST', url);
+  xhr.setRequestHeader('Content-Type', 'application/json');
+  xhr.send(JSON.stringify(data));
 }
 
+var buttons = document.querySelectorAll('.article__btn');
```
Comment (claireinez) [inline public/main.js]: Why are you using `var`s when everywhere else in the project there are `const`s?

File: src/api/guardian.js [hunk 3]
```diff
@@ -1,15 +1,30 @@
...
 };
 
-module.exports = guardianKey;
+const guardianItem = (id, cb) => {
+  const api = new Guardian(process.env.GUARDIAN_KEY, false);
+  api.item
+    .search(id, {
+      "show-fields": "bodyText"
+    })
+    .then(response => {
```
Comment (claireinez) [inline src/api/guardian.js]: This function could be written on one line

File: src/views/partials/header.hbs [hunk 4]
```diff
@@ -3,6 +3,6 @@
       <meta name="viewport" content="width=device-width,
       initial-scale=1.0">
       <meta http-equiv="X-UA-Compatible" content="ie=edge">
-      <link rel="stylesheet" href="/css/style.css">
+      <!-- <link rel="stylesheet" href="/css/style.css"> -->
```
Comment (claireinez) [inline src/views/partials/header.hbs]: If this is not needed, remove it

**Comment by sophieeejw** [comment]
@claireinez @Karyum waiting for the review :)

File: public/main.js [hunk 5]
```diff
@@ -1,14 +1,35 @@
...
+      callback(null, xhr.responseText);
+    }
+  };
+  xhr.open('POST', url);
+  xhr.setRequestHeader('Content-Type', 'application/json');
+  xhr.send(JSON.stringify(data));
 }
+
+
+var buttons = document.querySelectorAll('.article__btn');
```
Comment (claireinez) [inline public/main.js]: Why `var`?

File: public/main.js [hunk 6]
```diff
@@ -1,14 +1,35 @@
...
+    e.preventDefault();
+    var query = [btn.parentElement[0].value];
+    console.log(query);
+    var data = fetchPOST('/visualize', query, function(err, data) {
+      if (err) {
+        console.log(err);
+      }
+      else {
+        //This is the time to process the data using D3.js
+        console.log(data);
```
Comment (claireinez) [inline public/main.js]: Get rid of the console.log

**Comment by sophieeejw** [comment]
@MynahMarie

File: public/main.js (hunk 5 above)
Comment (prodionov) [reply public/main.js]: because we should use ES5 on the front end?

File: src/views/partials/header.hbs (hunk 4 above)
Comment (MynahMarie) [reply src/views/partials/header.hbs]: This will be needed once we start the CSS (which is what I'm working on now...)

File: public/main.js (hunk 6 above)
Comment (MynahMarie) [reply public/main.js]: We need the console log for now to help the people working on the data rendering.

File: src/router.js [hunk 7]
```diff
@@ -1,4 +1,9 @@
 const express = require("express");
+const { guardianKey, guardianItem } = require('./api/guardian');
```
Comment (prodionov) [inline src/router.js]: guardianKey should be guardianListing

File: src/router.js [hunk 8]
```diff
@@ -1,4 +1,9 @@
...
+const qs = require('querystring');
+const sentiment = require('sentiment');
+const polarity = require('polarity');
+const emotional = require('emotional');
 const router = express.Router();
 const guardianListing = require("./api/guardian")
```
Comment (prodionov) [inline src/router.js]: line 7 should be removed

File: src/api/guardian.js (hunk 3 above)
Comment (MynahMarie) [reply src/api/guardian.js]: I've changed it a little bit and not sure how to reduce it more... please check update.

File: src/router.js [hunk 9]
```diff
@@ -1,4 +1,9 @@
 const express = require("express");
+const { guardianKey, guardianItem } = require('./api/guardian');
+const qs = require('querystring');
+const sentiment = require('sentiment');
+const polarity = require('polarity');
+const emotional = require('emotional');
 const router = express.Router();
```
Comment (prodionov) [inline src/router.js]: this line should be removed

File: src/router.js (hunk 8 above)
Comment (prodionov) [inline src/router.js]: line 8 should be removed

**Comment by claireinez** [comment]
I also need to know which issue this is closing. Please edit the original PR description to include the line "Closes #<number>". "Closes" is a GitHub keyword and when the PR is merged, the corresponding issue will be closed.

File: public/main.js (hunk 5 above)
Comment (claireinez) [reply public/main.js]: Lol oops, I'm too used to using babel on the front end.

File: src/views/partials/header.hbs (hunk 4 above)
Comment (claireinez) [reply src/views/partials/header.hbs]: You can add it in when it's needed ;)

File: public/main.js (hunk 6 above)
Comment (claireinez) [reply public/main.js]: They can write the console.log in themselves 😃 . The comment here is sufficient as a TODO note

File: src/api/guardian.js (hunk 3 above)
Comment (claireinez) [reply src/api/guardian.js]: Is this update still to be pushed?

File: public/main.js (hunk 6 above)
Comment (MynahMarie) [reply public/main.js]: Ok cool will correct

**Comment by MynahMarie** [comment]
@claireinez Deleted the style cheat link and console log as requested :)

**Pull request merged**